scripts/opportunity-provider-limo/index.ts

```diff
@@ -0,0 +1,206 @@
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import * as fs from "fs";
+import { Connection, Keypair, PublicKey } from "@solana/web3.js";
+import * as anchor from "@coral-xyz/anchor";
+import * as limo from "@kamino-finance/limo-sdk";
+import { Decimal } from "decimal.js";
+import { getMintDecimals } from "@kamino-finance/limo-sdk/dist/utils";
+
+interface Token {
+  mint: PublicKey;
+  symbol: string;
+}
+
+interface OpportunityPair {
+  token1: Token;
+  token2: Token;
+  randomizeSides: boolean;
+  minAmountNotional: number;
+  maxAmountNotional: number;
+}
+
+function readFile<T>(path: string): T {
+  const data = fs.readFileSync(path, "utf8");
+  return JSON.parse(data) as T;
+}
+
+function loadOpportunities(path: string): OpportunityPair[] {
+  const opportunityPairs = readFile<OpportunityPair[]>(path);
+  return opportunityPairs.map((opportunity) => {
+    return {
+      token1: {
+        mint: new PublicKey(opportunity.token1.mint),
+        symbol: opportunity.token1.symbol,
+      },
+      token2: {
+        mint: new PublicKey(opportunity.token2.mint),
+        symbol: opportunity.token2.symbol,
+      },
+      randomizeSides: opportunity.randomizeSides,
+      minAmountNotional: opportunity.minAmountNotional,
+      maxAmountNotional: opportunity.maxAmountNotional,
+    };
+  });
+}
+
+const decimals: Record<string, number> = {};
+const prices: Record<string, number> = {};
+
+async function getDecimals(
+  connection: Connection,
+  token: Token
+): Promise<number> {
+  const index = token.mint.toBase58();
+  if (decimals[index] === undefined) {
+    decimals[index] = await getMintDecimals(connection, token.mint);
+  }
+
+  return decimals[index];
+}
+
+async function getPrice(connection: Connection, token: Token): Promise<number> {
+  const index = token.symbol;
+  if (prices[index] === undefined) {
+    const url = `https://api.binance.com/api/v3/ticker/price?symbol=${token.symbol}USDT`;
+    const response = await fetch(url);
+    const data = await response.json();
+    const price = parseFloat(data.price);
+    if (isNaN(price)) {
+      throw new Error(`Invalid price: ${data.price}`);
+    }
+    prices[index] = price;
+  }
+
+  const mintDecimals = await getDecimals(connection, token);
+  return prices[index] / Math.pow(10, mintDecimals);
+}
+
+async function createOpportunities(
+  skExecutor: Keypair,
+  limoClient: limo.LimoClient,
```

Comment:
i'd rather get the global_config and create the limoClient inside this function.